### PR TITLE
wrangler: fix darwin builds

### LINKED ISSUE #352287: Build failure: wrangler
### Steps To Reproduce

Wrangler package has been broken since PR https://github.com/NixOS/nixpkgs/pull/322573

This application should work just fine on all systems and did before the change.
There seem to be multiple issues with it:
- it's not a gui application, but currently depends on x11
- musl is in the dependencies, but is not normally necessary for the build
- patchelf fails on darwin (likely detects some library/binary as elf even though it isn't)

Also, some packages are removed even thought they seem necessary for workers deployment (like typescript).

In general it's a bit of a mess, even for linux-x86.

### Notify maintainers

@seanrmurphy @dezren39 @ryand56 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by viraptor**
Remove unnecessary deps, update platforms.

Closes #352287

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/wr/wrangler/package.nix [hunk 1]
```diff
@@ -30,15 +30,16 @@ stdenv.mkDerivation (finalAttrs: {
   buildInputs = [
     llvmPackages.libcxx
     llvmPackages.libunwind
-    musl
```
Comment (viraptor) [inline pkgs/by-name/wr/wrangler/package.nix:33]: Still not sure why it was here. A few packages mention it - for example:
```
lib/node_modules/.pnpm/@types+node@20.8.3/node_modules/@types/node/ts4.8/fs.d.ts:         * On Linux, when Node.js is linked against musl libc, the procfs file system must
```
But it doesn't look like the nixos version is using musl at all.

**Review by ryand56 — COMMENTED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352305`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>wrangler</li>
  </ul>
</details>

Seems like ofborg is having an issue with patchelf involving musl still.

**Comment by viraptor** [comment]
One of the pre-built packages partially depends on musl.
The options are:
1. Remove the musl version since it's not used (or skip patching it) 
2. Restore node-pty instead of using the pre-built one. This would be helpful for BSD and RISC in the future, but may require some ongoing patching.

**Review by ryand56 — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352305`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>wrangler</li>
  </ul>
</details>

I'm not sure what patching will need to be done to move away from musl.

**Review by dezren39 — APPROVED** [review]

**Comment by viraptor** [comment]
We can address musl separately later. It's going to be a bit of work.

**Pull request merged**